libvast/include/vast/dump_operator.hpp

```diff
@@ -0,0 +1,46 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/logical_operator.hpp"
+
+#include <vast/plugin.hpp>
+
+namespace vast {
+
+/// The logical operator for dumping data that will have to be joined later
+/// during pipeline execution.
+class dump_operator : public logical_operator<chunk_ptr, void> {
```

Comment:
@jachris found a nice approach in #3016 by creating a single file `libvast/builtins/from_read.cpp` that calls `VAST_REGISTER_PLUGIN` twice, avoiding the need for the logic shared between the from and read operators to leak into libvast itself. I would recommend you follow the same approach here for the write and to operators.

Reply:
I moved everything into `libvast/builtins/operators/write_to.cpp` in https://github.com/tenzir/vast/pull/3024/commits/da051bbe03ddfcb5be49854d9fabdff6ffa857c2